pkg/pillar/zedcloud/deferred.go

```diff
@@ -192,25 +190,6 @@ func (ctx *DeferredContext) handleDeferred(event time.Time,
 			}
 			item.buf = nil
 			sent++
-
-			if sendOne {
-				exit = true
-				break
-			}
-
-			if time.Since(event) > maxTimeToHandleDeferred {
-				log.Warnf("handleDeferred: took too long time %v",
-					time.Since(event))
-				exit = true
-				break
-			}
-
-			// XXX sleeping in main thread
-			if len(reqs)-sent != 0 && spacing != 0 {
```

Comment:
Let's check the history of why spacing was introduced.

Reply:
I just see the commit https://github.com/lf-edge/eve/commit/80cedef4f4ad92da2966d5e9ca56d90b646389df without any bug motivating the change. It could be to reduce sending rate towards the controller. So I think it is fine to drop the spacing now.

Reply:
Either we try to be smarter than TCP stack with all its congestion control, slow-start etc, either we save controller from frequent requests. In both cases looks very strange to me.

Reply:
One could argue that we might want to limit sending 100 info messages all at once if we have 100 objects (app instances, blobs, etc) to report, but that means the controller isn't very robust against load. And if we really want to limit, then we need some binary exponential backoff and not a fixed delay!

But 100 is small. FWIW loguploader has some rate limiting, but it could have many gigabytes of logs to send if the network has been out for a while.

Hence I have no issue with dropping this delay; just wanted to look at why it was introduced and explain my thinking.